Mathlib/Data/Finsupp/Fin.lean

```diff
@@ -23,16 +27,35 @@ noncomputable section
 
 namespace Finsupp
 
-variable {n : ℕ} (i : Fin n) {M : Type*} [Zero M] (y : M) (t : Fin (n + 1) →₀ M) (s : Fin n →₀ M)
+variable {n : ℕ} {M : Type*} [Zero M]
 
 /-- `tail` for maps `Fin (n + 1) →₀ M`. See `Fin.tail` for more details. -/
 def tail (s : Fin (n + 1) →₀ M) : Fin n →₀ M :=
   Finsupp.equivFunOnFinite.symm (Fin.tail s)
 
+/-- `init` for maps `Fin n →₀ M`. See `Fin.init` for more details. -/
+def init (s : Fin (n + 1) →₀ M) : Fin n →₀ M :=
+  Finsupp.equivFunOnFinite.symm (Fin.init s)
+
+/-- `removeNth` for maps `Fin n →₀ M`. See `Fin.removeNth` for more details. -/
+def removeNth (p : Fin (n + 1)) (s : Fin (n + 1) →₀ M) : Fin n →₀ M :=
+  Finsupp.equivFunOnFinite.symm (Fin.removeNth p s : Fin n → M)
+
 /-- `cons` for maps `Fin n →₀ M`. See `Fin.cons` for more details. -/
 def cons (y : M) (s : Fin n →₀ M) : Fin (n + 1) →₀ M :=
   Finsupp.equivFunOnFinite.symm (Fin.cons y s : Fin (n + 1) → M)
 
+/-- `snoc` for maps `Fin n →₀ M`. See `Fin.snoc` for more details. -/
+def snoc (s : Fin n →₀ M) (y : M) : Fin (n + 1) →₀ M :=
+  Finsupp.equivFunOnFinite.symm (Fin.snoc s y : Fin (n + 1) → M)
+
+/-- `insertNth` for maps `Fin n →₀ M`. See `Fin.insertNth` for more details. -/
+def insertNth (p : Fin (n + 1)) (y : M) (s : Fin n →₀ M) : Fin (n + 1) →₀ M :=
+  Finsupp.equivFunOnFinite.symm (Fin.insertNth p y s : Fin (n + 1) → M)
+
+variable (p : Fin (n + 1)) (y : M) (t : Fin (n + 1) →₀ M) (s : Fin n →₀ M) (i : Fin n)
```

Comment:
When you removed these variables from the other variable line above, you reordered them. Can you preserve the order, or at least explain why this order is preferable? This changed the variable order in `cons_succ` (and probably other lemmas) a few lines below this.